### PR TITLE
Fix small precision issues & upgrade go.mod version to 1.17

### PR DESCRIPTION
**Pull request by SkillingX**
Fix small precision issues & upgrade go.mod version to 1.17

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,12 +1,19 @@
 module github.com/eteu-technologies/near-api-go
 
-go 1.16
+go 1.17
```
Comment (mikroskeem) [inline go.mod:3]:
```suggestion
go 1.18
```

Might as well bump to 1.18 now.

File: pkg/types/balance.go [hunk 2]
```diff
@@ -7,11 +7,13 @@ import (
 	"math/big"
 
 	uint128 "github.com/eteu-technologies/golang-uint128"
+	"github.com/shopspring/decimal"
```
Comment (mikroskeem) [inline pkg/types/balance.go:10]: Not very happy with introducing this dependency without an explanation

**Review by mikroskeem — CHANGES_REQUESTED** [review]
It's not very obvious what's wrong and what does this fix. Can you please elaborate what's wrong currently?

Can you please add unit tests as well to the proposed fix?

File: go.mod (hunk 1 above)
Comment (SkillingX) [reply go.mod:3]: This is just to upgrade the version of go by the way. As for 1.17 or 1.18, it has no essential impact.

File: pkg/types/balance.go (hunk 2 above)
Comment (SkillingX) [reply pkg/types/balance.go:10]: This dependency library is very reliable. Most of the conversion and calculation operations involving floating-point numbers in our project are solved by using it.

File: go.mod (hunk 1 above)
Comment (mikroskeem) [reply go.mod:3]: Yeah so let's bump it to 1.18 then.

File: pkg/types/balance.go (hunk 2 above)
Comment (mikroskeem) [reply pkg/types/balance.go:10]: But what problem does it solve here?

**Comment by mikroskeem** [comment]
Thank you for your effort. PR #28 supersedes this and also adds unit tests to detect regressions in future.

**Pull request closed without merging**